### ISSUE TITLE
Error: The specified module could not be found

### ISSUE THREAD

**Issue by DethRaid** [microsoft/CsWinRT#407]
I'm trying to use CsWinRT in my project. I have a bunch of C++ code, then I expose specific C++ classes as WinRT runtime classes. I then generate C# files for all my WinRT interfaces using CsWinRT, and have a C# application that calls into the C++ code by way of the WinRT runtime classes

I have my build system set up, I'm generating the C# files, everything is groovy... until I run my code. At runtime, when the .NET runtime first loads one of the WinRT runtime classes, I get an exception with the message "The specified module could not be found"

My assumption has been that the CoreCLR runtime is failing to find the DLL with my C++ classes. I've tried giving it a few different names, copying it into the directory with my C# application, using a `app.manifest` to specifically expose my WinRT runtime classes... nothing works. I get exceptions about not being able to load a module no matter what I do

My C# application lives [here](https://github.com/DethRaid/SanityEngine/tree/winrt/Sanity.Editor), the interop DLL is in [this folder](https://github.com/DethRaid/SanityEngine/tree/winrt/SanityEngine.NET), my C++ is in [here](https://github.com/DethRaid/SanityEngine/tree/winrt/SanityEngine)

**Comment by AdamBraden** [comment]
@angelazhangmsft - can you take a look. Please double check the VCRT Forwarders, etc.

**Comment by Scottj1s** [comment]
This would be a good topic for a wiki page - troubleshooting activation errors

**Comment by DethRaid** [comment]
Are there some troubleshooting steps I could try while yalls write up the wiki page?

**Comment by manodasanW** [comment]
@DethRaid I took a look at your project.  I saw that the VCRT Forwarders reference was on your [component project](https://github.com/DethRaid/SanityEngine/blob/bb9900855e95bc3476b6bea10edae9c395eff56e/SanityEngine/SanityEngine.vcxproj#L789).  The VCRT forwarders reference is intended to be applied on the application project.  So I believe if you move that reference to the application project, it should work.  As a confirmation, once you move it, you should also see the VCRT forwarders for the architecture being built copied in the bin folder for your application alongside your application binaries.  Let us know if that fixes it.

**Comment by DethRaid** [comment]
I'm currently blocked by #424 . As soon as that's resolved, I'll test out that fix

**Comment by DethRaid** [comment]
#424 got resolved, but I have a new error - #436

**Comment by DethRaid** [comment]
After some truly wonderful work by all yalls to resolve my various compile issues, I was finally able to run my application after moving the VCRT forwarders reference to my application project!

I don't see any VCRT forwarders in my application's bin directory. I'm not entirely sure what I'm supposed to be looking for, though. I uploaded a screenshot of my `bin` directory to [here](https://imgur.com/qAXwn5L)

When I run my application I get the same error as before, where it can't load my interop assembly

I've updated the code in my GitHub with all my recent changes

**Comment by Scottj1s** [comment]
@DethRaid, glad for the progress.  We'll have a look at the latest code asap.

**Comment by manodasanW** [comment]
@DethRaid it does look like the VCRT forwarders are being copied based on your screenshot.  They are the binaries which start with vc like vcruntime140_app.dll.  I took another look at your branch and tried to build it to get a repro, but it seems I am running into various dependency issues and various absolute path issues.  Is it possible for you to provide an isolated repro that I can take a look at.

And to confirm are you running your application by running the exe (Sanity.Editor.exe)?

**Comment by DethRaid** [comment]
This issue is almost four weeks old. Before opening this issue, I spent 3-4 weeks on my own learning WinRT and MIDL, and integrating them into my project. That's 7-8 weeks I've spent trying to plug my C# and C++ together using WinRT

This weekend I went back to the drawing board. I read up on P/Invoke, realized that it was exactly what I needed, and in the last few days I've gotten farther with P/Invoke than I did in a few weeks of working with WinRT

I'm closing this issue as obsolete. Everyone I've interacted with when trying to use C#/WinRT has been incredibly helpful. Yalls are awesome and I wish your time could have been spent with someone who would ultimately succeed at using C#/WinRT. At the end of the day, though, I've been _far_ more successful using P/Invoke for C++/C# interop than C#/WinRt
